### PR TITLE
feat: add C implementation for `@stdlib/stats/base/dists/pareto-type1/median`

### PR DESCRIPTION
**Pull request by rautelaKamal**
Ref: #3861.

## Description

This PR adds a double-precision C implementation for `@stdlib/stats/base/dists/pareto-type1/median`, following the pattern established by the arcsine CDF implementation (PR #3354).

### Changes

- Add C header file (`include/stdlib/stats/base/dists/pareto-type1/median.h`)
- Implement `stdlib_base_dists_pareto_type1_median` in C for double-precision
- Create Node-API addon bridge using `STDLIB_MATH_BASE_NAPI_MODULE_DD_D` macro
- Add GYP build configuration (`binding.gyp`, `include.gypi`, `manifest.json`)
- Create `lib/native.js` dispatcher for JavaScript integration
- Update `package.json` to enable gypfile
- Add comprehensive native test suite (`test/test.native.js`) with tryRequire pattern
- Add JavaScript native benchmark (`benchmark/benchmark.native.js`)
- Add C benchmark (`benchmark/c/benchmark.c`)
- Add C example (`examples/c/example.c`)
- Update README with C API documentation

### Testing

- Native test suite mirrors existing `test/test.js` with all edge cases
- Uses standard `tryRequire`/`skip` pattern for graceful degradation

## Related Issues

Does this pull request have any related issues?

This pull request has the following related issues:

- [#3861](https://github.com/stdlib-js/stdlib/issues/3861)
- [#3355](https://github.com/stdlib-js/stdlib/issues/3355)

## Questions

Any questions for reviewers of this pull request?

No.

## Other

Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

Please ensure the following tasks are completed before submitting this pull request.

- [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)

### CODE REVIEW

File: lib/node_modules/@stdlib/stats/base/dists/pareto-type1/median/benchmark/benchmark.native.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+var median = tryRequire( resolve( __dirname, './../lib/native.js' ) );
+var opts = {
+	'skip': ( median instanceof Error )
+};
+
+
+// MAIN //
+
+bench( pkg+'::native', opts, function benchmark( b ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/pareto-type1/median/benchmark/benchmark.native.js]: Missing `format` require. Add `var format = require( '@stdlib/string/format' );` after the `pkg` require, and use `format( '%s::native', pkg )` on line 42 instead of string concatenation. `format` is the preferred approach for string concatenation in the project.

File: lib/node_modules/@stdlib/stats/base/dists/pareto-type1/median/benchmark/benchmark.native.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	var alpha;
+	var beta;
+	var o;
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/pareto-type1/median/benchmark/benchmark.native.js]: Convention is to name this variable `opts` (re-declared via `var` inside the function scope), matching the pattern in other benchmark files. See e.g. `@stdlib/stats/base/dists/beta/mean/benchmark/benchmark.native.js`.

```suggestion
	var opts;
```

**Comment by Witty-31-06** [comment]
Hi a [PR](https://github.com/stdlib-js/stdlib/pull/10079) is already raised on this issue which is pending review.

**Comment by Neerajpathak07** [comment]
@rautelaKamal Just a little heads-up there is an existing PR resolving this issue. Ref:- #10080
